src/goto-symex/symex_goto.cpp

```diff
@@ -328,8 +329,14 @@ void goto_symext::phi_function(
       if(p_it!=goto_state.propagation.values.end())
         goto_state_rhs=p_it->second;
       else
-        to_ssa_expr(goto_state_rhs).set_level_2(
-          goto_state.level2_current_count(l1_identifier));
+      {
+        unsigned l2=goto_state.level2_current_count(l1_identifier);
+        // do not merge when we don't have a local symbol in the goto
+        // state; those should have been filtered out already in
+        // goto_symext::symex_assign_symbol
+        CHECK_RETURN(l2!=0 || it->get_level_1().empty());
```

Comment:
Yes, it seems I've got quite a bit more work to do here. I've figured out some of the cases (function parameters), with others still to be done. My apologies that this is taking much longer than expected.